packages/flutter/lib/src/material/tabs.dart

```diff
@@ -568,11 +568,134 @@ class _TabBarScrollController extends ScrollController {
 ///
 /// Uses values from [TabBarTheme] if it is set in the current context.
 ///
-/// To see a sample implementation, visit the [TabController] documentation.
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// This sample shows the implementation of [TabBar] with a [DefaultTabController]
+/// and a [TabBar] consisting of three [Tab] widgets with an [Icon] as a child.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///    return DefaultTabController(
+///      initialIndex: 1,
+///      length: 3,
+///      child: Scaffold(
+///        appBar: AppBar(
+///          title: Text("TabBar Widget"),
+///          bottom: TabBar(
+///            tabs: [
+///              Tab(
+///                icon: Icon(
+///                  Icons.cloud_outlined,
+///                ),
+///              ),
+///              Tab(
+///                icon: Icon(
+///                  Icons.beach_access_sharp,
+///                ),
+///              ),
+///              Tab(
+///                icon: Icon(
+///                  Icons.brightness_5_sharp,
+///                ),
+///              ),
+///            ],
+///          ),
+///        ),
+///        body: TabBarView(
+///          children: [
+///            Center(
+///              child: Text(
+///                "It's cloudy here",
+///              ),
+///            ),
+///            Center(
+///              child: Text(
+///                "It's rainy here",
+///              ),
+///            ),
+///            Center(
+///              child: Text(
+///                "It's sunny here",
+///              ),
+///            ),
+///          ],
+///        ),
+///      ),
+///    );
+///  }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// [TabBar] can also be implmented by using a [TabController] which provides more options
+/// to control the behaviour of the [TabBar] and [TabBarView], this can be used instead of
+/// a [DefaultTabController], below is a sample for this.
```

Comment:
```suggestion
/// to control the behavior of the [TabBar] and [TabBarView]. This can be used instead of
/// a [DefaultTabController], demonstrated below.
```